File: backend/app/schemas/content.py
import uuid
from datetime import datetime

from pydantic import BaseModel, computed_field


class PublicCmsBlock(BaseModel):
    key: str
    page: str
    section: str | None
    label: str | None
    value: str | None
    display_order: int


class PublicTeamMember(BaseModel):
    id: uuid.UUID
    name: str
    role: str | None
    bio: str | None
    photo_url: str | None
    display_order: int

    model_config = {"from_attributes": True}


_WORDS_PER_MINUTE = 200


def _read_time_minutes(content_html: str | None) -> int:
    if not content_html:
        return 1
    import re

    text = re.sub(r"<[^>]+>", " ", content_html)
    word_count = len(text.split())
    return max(1, -(-word_count // _WORDS_PER_MINUTE))  # ceil division


class PublicBlogPostSummary(BaseModel):
    id: uuid.UUID
    title: str
    category: str | None
    tag_key: str | None
    author_name: str | None
    excerpt: str | None
    featured: bool
    featured_image_url: str | None
    tags: list[str]
    views: int
    created_at: datetime


class PublicBlogPostDetail(PublicBlogPostSummary):
    content_html: str | None

    @computed_field
    @property
    def read_time_minutes(self) -> int:
        return _read_time_minutes(self.content_html)

File: frontend/src/lib/types/content.ts
// Mirrors backend/app/schemas/content.py exactly — keep these two in sync.

export interface PublicCmsBlock {
  key: string;
  page: string;
  section: string | null;
  label: string | null;
  value: string | null;
  display_order: number;
}

export interface PublicTeamMember {
  id: string;
  name: string;
  role: string | null;
  bio: string | null;
  photo_url: string | null;
  display_order: number;
}

export interface PublicBlogPostSummary {
  id: string;
  title: string;
  category: string | null;
  tag_key: string | null;
  author_name: string | null;
  excerpt: string | null;
  featured: boolean;
  featured_image_url: string | null;
  tags: string[];
  views: number;
  created_at: string;
}

export interface PublicBlogPostDetail extends PublicBlogPostSummary {
  content_html: string | null;
  read_time_minutes: number;
}

// Mirrors backend/app/schemas/admin_contact.py's ContactSubmissionCreate
// (shared between the public /contact submission and admin read/reply).
export interface ContactSubmissionCreate {
  name: string;
  email: string;
  organization?: string | null;
  subject: string;
  message: string;
}
